src/cron-job.ts

```diff
@@ -0,0 +1,251 @@
+   */
+  readonly jobProperties: JobProps;
+
+  /**
+   * Specifies the time in which the job would run again.
+   *
+   * @default - Runs at every minute.
+   */
+  readonly schedule?: CronOptions;
+
+  /**
+   * Specifies the timezone for the job. This helps aligining the schedule to follow the specified timezone.
+   *
+   * @default - Timezone of kube-controller-manager process.
+   */
+  readonly timeZone?: string;
+
+  /**
+   * Specifies the concurrency policy for the job.
+   *
+   * @default - Allow
+   */
+  readonly concurrencyPolicy?: ConcurrencyPolicy;
+
+  /**
+   * Specifies the time by which a recurring job should be scheduled after it has passed its expected scheduling time.
+   * Missed jobs are counted as failed ones.
+   *
+   * @default - There is no deadline.
+   */
+  readonly startingDeadline?: Duration;
+
+  /**
+   * Specifies if the recurring job should be suspended.
+   *
+   * @default - false.
+   */
+  readonly suspend?: boolean;
+
+  /**
+   * Specifies the number of successful jobs history retained.
+   *
+   * @default - 3.
+   */
+  readonly successfulJobsRetained?: number;
+
+  /**
+   * Specifies the number of failed jobs history retained.
+   *
+   * @default - 1.
+   */
+  readonly failedJobsRetained?: number;
+}
+
+/**
+ * A CronJob is responsible for creating a Job and scheduling it based on provided scheduled. This helps running Jobs in a recurring manner.
+ */
+export class CronJob extends workload.Workload {
+  /**
+   * Properties of the recurring `Job`.
+   */
+  public readonly jobProperties: JobProps;
+
+  /**
+   * Schedule for which to run the recurring job. By default, the job would be scheduled to run every minute.
+   */
+  public readonly schedule?: CronOptions;
+
+  /**
+   * Timezone in which the recurring job is scheduled.
+   */
+  public readonly timeZone?: string;
+
+  /**
+   * Concurrency policy for the recurring job. This specifies how the job scheduling would be handled based on which policy is selected.
+   */
+  public readonly concurrencyPolicy?: string;
+
+  /**
+   * Deadline to start next instance of job after the expected scheduling time. The job is considered as failed if it misses this deadline.
+   */
+  public readonly startingDeadline?: Duration;
+
+  /**
+   * Flag that tells to suspend the upcoming executions of the job.
+   */
+  public readonly suspend?: boolean;
+
+  /**
+   * The number of successful jobs to retain.
+   */
+  public readonly successfulJobsRetained?: number;
+
+  /**
+   * The number of failed jobs to retain.
+   */
+  public readonly failedJobsRetained?: number;
+
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  /**
+   * Represents the resource type.
+   */
+  public readonly resourceType = 'cronjob';
+
+  constructor(scope: Construct, id: string, props: CronJobProps) {
+    super(scope, id, {
+      restartPolicy: RestartPolicy.ON_FAILURE,
```

Comment:
I imagined it would help with job completion but now that I think about it, even if this restarts, I am not sure the cronjob would still keep track of this particular completion and would probably mark it as failed or reschedule it.